spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java

```diff
@@ -908,7 +908,7 @@ public JSONObject changeLibraryDilutionRow(HttpSession session, JSONObject json)
                 + (isStringEmptyOrNull(dilution.getIdentificationBarcode()) ? "" : dilution.getIdentificationBarcode()) + "'/>");
       }
       if (dilution.getLibrary().getLibraryAdditionalInfo() != null && dilution.getLibrary().getLibraryAdditionalInfo().getPrepKit() != null
-          && dilution.getLibrary().getLibraryAdditionalInfo().getPrepKit().getId() != null && json.getBoolean("detailedSample")) {
+          && json.getBoolean("detailedSample")) {
```

Comment:
Please revert and refactor, or else we risk NPEs! It should be the LibraryAdditionalInfo's KitDescriptor's ID that is not null.